From 4f09c4d4c1ad2026346a98e63f0e13155d0f0487 Mon Sep 17 00:00:00 2001
From: Dusty Mabe <dusty@dustymabe.com>
Date: Mon, 25 Nov 2024 16:29:10 -0500
Subject: [PATCH 2/5] osbuild/loop: make the loop device if missing

A few times during development I saw an error where the loop
device wasn't getting created. Maybe it was some weird state
issue with my system (i.e. loopback devices are global), or
maybe not. Either way maybe it won't hurt to create it if
it doesn't exist.
---
 osbuild/loop.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/osbuild/loop.py b/osbuild/loop.py
index ec6d3619..b768af22 100644
--- a/osbuild/loop.py
+++ b/osbuild/loop.py
@@ -126,6 +126,12 @@ class Loop:
             if not dir_fd:
                 dir_fd = os.open("/dev", os.O_DIRECTORY)
                 stack.callback(lambda: os.close(dir_fd))
+            # If the loopdev didn't show up for some reason let's
+            # create it manually
+            try:
+                os.stat(self.devname, dir_fd=dir_fd)
+            except FileNotFoundError:
+                self.mknod(dir_fd)
             self.fd = os.open(self.devname, os.O_RDWR, dir_fd=dir_fd)
 
         info = os.stat(self.fd)
-- 
2.47.0

